test/LocalizationWebsite/StartupResourcesInClassLibrary.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Localization;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Logging;
+
+namespace LocalizationWebsite
+{
+    public class StartupResourcesInClassLibrary
+    {
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddLocalization(options => options.ResourcesPath = "Resources");
+        }
+
+        public void Configure(
+            IApplicationBuilder app,
+            ILoggerFactory loggerFactory,
+            IStringLocalizerFactory stringLocalizerFactory,
+            IStringLocalizer<ResourcesClassLibraryNoAttribute.Model> noAttributeStringLocalizer,
+            IStringLocalizer<ResourcesClassLibraryWithAttribute.Model> withAttributeStringLocalizer)
+        {
+            loggerFactory.AddConsole(minLevel: LogLevel.Warning);
+
+            var supportedCultures = new List<CultureInfo>()
+            {
+                new CultureInfo("en-US"),
+                new CultureInfo("fr-FR")
+            };
+
+            app.UseRequestLocalization(new RequestLocalizationOptions
+            {
+                DefaultRequestCulture = new RequestCulture("en-US"),
+                SupportedCultures = supportedCultures,
+                SupportedUICultures = supportedCultures
+            });
+
+            var noAttributeAssembly = typeof(ResourcesClassLibraryNoAttribute.Model).GetTypeInfo().Assembly;
+            var noAttributeName = new AssemblyName(noAttributeAssembly.FullName).Name;
+            var noAttributeNameStringLocalizer = stringLocalizerFactory.Create(
+                nameof(ResourcesClassLibraryNoAttribute.Model),
+                noAttributeName);
+
+            var withAttributeAssembly = typeof(ResourcesClassLibraryWithAttribute.Model).GetTypeInfo().Assembly;
+            var withAttributeName = new AssemblyName(withAttributeAssembly.FullName).Name;
+            var withAttributeNameStringLocalizer = stringLocalizerFactory.Create(
+                nameof(ResourcesClassLibraryWithAttribute.Model),
+                withAttributeName);
```

Comment:
We create `IStringLocalizer`'s with that overload above this on L43 and L44, then use them on L62 and L66. They also have unit tests in ResourceManagerStringLocalizerFactoryTest.